68: Support for dry-run to print undefined steps

TitouanVanBelle opened the pull request:
This pull request allow users to use the dry run feature described in #68 the following way
```
[Cucumberish instance].dryRun = YES;
[Cucumberish instance].dryRunLanguage = CCILanguageObjectiveC;
```

The results are printed in the logs the following way

*dryRunLanguage = CCILanguageObjectiveC*

```
2017-06-16 16:52:22.892 xctest[39431:10863180] 

Dry Run Results
=====================================================
Found the following undefined steps:

Given(@"^a Given statement$", ^(NSArray<NSString *> *args, NSDictionary *userInfo) {

});

And(@"^an And statement$", ^(NSArray<NSString *> *args, NSDictionary *userInfo) {

});

=====================================================
```

*dryRunLanguage = CCILanguageSwift*

```
2017-06-16 16:52:22.892 xctest[39431:10863180] 

Dry Run Results
=====================================================
Found the following undefined steps:

Given("^a Given statement$") { (args, userInfo) in

}

And("^an And statement$") { (args, userInfo) in

}

=====================================================
```

When all steps are defined, the following is printed

```
Dry Run Results
=====================================================

All the steps are defined

=====================================================
```

TitouanVanBelle commented:
Sure. Any idea what kind of tests would fit better for this feature? Do you think it makes sense to add tests to the CucumberishTests target for this?

brentleyjones commented:
Yeah, add tests to CucumberishTests target. After that I'll get to this after the 4th of July. Busy at work and then a small vacation.

TitouanVanBelle commented:
I added some test to the CucumberishFeatureDefinition target because the step definitions form the CucumberTests target are using `@"(.*)"` which makes it hard to have undefined steps.

I had to add a log manager to assert that the logs were properly printed. It might a bit overcomplicated for the task but it does the job.

I also had some problem since the steps are not executed when `dryRun` is enabled so I had to do the assertion in the `afterFinish` block. The problem with this solution is that it will fail the last scenario executed if the assert fails. So the dryRun scenario should always be executed last and that's why I added numbers to the feature names.

brentleyjones commented on Cucumberish/Core/Models/CCIStep.h:
```diff
@@ -82,4 +87,9 @@ typedef NS_ENUM(NSInteger,CCIStepStatus) {
\ No newline at end of file
+
+/**
+ @return a string composed of the keyword and the text of the tep
```
Typo: `tep`

brentleyjones commented:
Thanks @TitouanVanBelle!
